cli/js/io.ts

```diff
@@ -104,3 +107,27 @@ export function toAsyncIterator(r: Reader): AsyncIterableIterator<Uint8Array> {
     },
   };
 }
+
+export function iterSync(
+  r: SyncReader,
+  bufSize?: number
+): IterableIterator<Uint8Array> {
+  const b = new Uint8Array(bufSize ?? 1024);
+  return {
+    [Symbol.iterator](): IterableIterator<Uint8Array> {
+      return this;
+    },
+
+    next(): IteratorResult<Uint8Array> {
+      const result = r.readSync(b);
+      if (result === EOF) {
+        return { value: new Uint8Array(), done: true };
+      }
+
+      return {
+        value: b.subarray(0, result),
```

Comment:
Is it correct to use this method? It doesn't copy, but I'm not sure.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/subarray#Description

Reply:
Yeah, it doesn't copy, it creates a new view using the same buffer

Reply:
Okay then this code will definitely not work as expected:
```js
const buffers = [...Deno.iterSync(Deno.openSync("temp"), 1)];

for (const b of buffers) {
  Deno.stdout.writeSync(b);
}
```
If it's required for efficiency, that definitely warrants a comment telling people to process a chunk before getting the next. But either way there'll be surprises for people refactoring.

Reply:
Nice one!

Here's my take:
```
   * Iterator uses internal buffer of fixed size for efficiency returning 
   * a view on that buffer on each iteration. It it therefore callers 
   * responsibility to copy contents of the buffer if needed; otherwise 
   * next iteration will overwrite contents of previously returned chunk.
```

@nayeemrmn could you provide an example?

Reply:
I think that comment suffices... hopefully people give it a read before trying to extract it to an array 🤞.